.github/workflows/common-java-gradle-release.yml

```diff
@@ -0,0 +1,108 @@
+# This is a basic release workflow for Octopus gradle Java modules with publishing to Sonatype Nexus and/or image registry
+name: Gradle Release
+
+on:
+  workflow_call:
+    inputs:
+      java-version:
+        required: false
+        type: string
+        default: '8'
+      docker-image:
+        description: 'Docker image name'
+        required: false
+        type: string
+        default: ''
+      increment-version-level:
+        description: 'One of: {major, premajor, minor, preminor, patch, prepatch, prerelease}'
+        required: false
+        type: string
+        default: patch
+    secrets:
+      OSSRH_USERNAME:
+        required: true
+      OSSRH_TOKEN:
+        required: true
+      GPG_PASSPHRASE:
+        required: true
+      GPG_PRIVATE_KEY:
+        required: true
+
+jobs:
+  prepare-build-publish-release:
+    runs-on: ubuntu-latest
+    environment: Prod
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get latest tag
+        id: tag_version
+        uses: oprypin/find-latest-tag@v1
+        with:
+          repository: ${{ github.repository }}
+          regex: '^v([0-9]+)\..*'
+
+      - name: Prepare new tag version
+        uses: actions-ecosystem/action-bump-semver@v1
+        id: new-tag-version
+        with:
+          current_version: ${{ steps.tag_version.outputs.tag }}
+          level: ${{ inputs.increment-version-level }}
+
+      - name: Get new version
+        uses: octopusden/octopus-base/.github/actions/get-version@v1.0
+        id: new-version
+        with:
+          tag-value: ${{ steps.new-tag-version.outputs.new_version }}
+          exit-if-empty: true
+
+      - run: |
+          echo "BUILD_VERSION=${{ steps.new-version.outputs.version }}" >> $GITHUB_ENV
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ inputs.java-version }}
+          distribution: 'adopt'
+          cache: 'gradle'
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Gradle build
+        if: ${{ inputs.docker-image == '' }}
+        run: ./gradlew build -Pversion=${{ env.BUILD_VERSION }} --info -s
+
+      - name: Gradle build docker image
+        if: ${{ inputs.docker-image }}
+        run: ./gradlew build dockerBuildImage -x test -Ppublishing.docker.registry=ghcr.io/octopusden -Pdocker.registry=docker.io -Pversion=${{ env.BUILD_VERSION }} --info -s
+
+      - name: Log in to Docker Registry
+        if: ${{ inputs.docker-image }}
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push to docker registry
+        if: ${{ inputs.docker-image }}
+        run: docker push ghcr.io/octopusden/${{ inputs.docker-image }}:${{ env.BUILD_VERSION }}
+
+      - name: Publish to Sonatype Nexus
```

Comment:
Seems bug.
We are building gradle artifact above ` if: ${{ inputs.docker-image == '' }}` only.
But trying to publish it here regardless of that.

Moreover: I prefer to split docker image build and gradle java build.

Reply:
We are building docker image in the step "Gradle build docker image", which has the same condition 
if: ${{ inputs.docker-image }}
I agree, but I didn't find a way to avoid code duplication

Reply:
OK for code duplication, but I am still not sure we will not fail here because `if: inputs.docker-image` then we do *NOT* build an artifact to deploy to MVN-compatible storage (see `if` condition on step *Gradle Build* above).
This case we have no artifact to deploy to but have a docker image, which is deployed in the step *Push to docker registry* above. What will happen here? Failure?

Reply:
> I prefer to split docker image build and gradle java build
А как ты разделяешь их? Я не вижу простого способа собрать java билд, а потом отдельно собрать docker образ из этой библиотеки. Просто сразу взять и использовать плагин сборки образа для spring приложения